remove instruction to checkout staging on local setup

mikedotexe opened the pull request:
Very simple, as the title says. There must have once been a time where it made sense to checkout staging, but I've been instructed to remove this.

damons commented:
Since we've changed the branching and code promotion scheme for nearcore, we need to make sure we instruct users to use our most stable network and code by default.  This will be the `stable` branch.

mikedotexe commented:
> Since we've changed the branching and code promotion scheme for nearcore, we need to make sure we instruct users to use our most stable network and code by default. This will be the `stable` branch.

goood catch. changed

mikedotexe commented:
just had a user get confused [about this on Discord](https://discordapp.com/channels/490367152054992913/490367152054992917/682799478507569161)
Going to merge this
